### ISSUE TITLE
Extend PyTorch-Based GPU Acceleration to Full Scan Workflows

### ISSUE THREAD

**Issue by GianmarcoMidena** [Probe-Particle/PPSTM#58]
## Motivation

The current PyTorch backend accelerates individual `dI/dV` evaluations, but higher-level PPSTM scan modes still call `dI/dV` repeatedly from Python.

This affects:

- `states` scans in `STMutils.run_stm_scan(...)`
- `STM(...)` scans in `ProbeSTM.py`
- `MSTM(...)` / voltage scans in `ProbeSTM.py`

As a result, PyTorch runs repeatedly move data between CPU and GPU, recreate tensors, launch many separate computations, and then stack or integrate results on the CPU.

A PyTorch-native full-scan path would reduce this overhead and would also make the code more useful for Machine Learning workflows, where gradients through PPSTM simulations may be needed.

## Current Behavior

`ProbeSTM.py` supports PyTorch for a single `dI/dV` calculation:
```python
dIdV(..., backend=DidvBackend.PYTORCH)
```
However, `STM(...)` still loops over voltage steps:
```python
for v_ in np.linspace(0., V, nV):
    i_ = dIdV(..., backend=backend)
    cur += i_
```
`MSTM(...)` also loops over voltage steps and then integrates on the CPU:
```python
for v_ in np.arange(Vmin, Vmax + 0.001, dV):
    i_ = dIdV(..., backend=backend)
    dIdVs = np.append(dIdVs, np.array([i_]), axis=0)
```
The `states` scan path in `STMutils.py` has the same issue:
```python
didvs = []
for state in states:
    didv = PS.dIdV(
        V=state,
        **didv_args,
        **tip_coefficients,
    )
    didvs.append(didv)

didv = np.stack(didvs, axis=0)
```
With the PyTorch backend, each of these calls creates a back-and-forth CPU/GPU workflow instead of keeping the full scan on the GPU.

## Goal

Add PyTorch-based GPU acceleration for full PPSTM scan workflows.

The scan-level implementation should evaluate voltage/state batches in one PyTorch tensor workflow instead of repeatedly dispatching independent single-voltage `dI/dV` calls from Python.

Shared inputs such as eigenenergies, tip positions, atom positions, coefficients, and tip orbital coefficients should stay on the selected PyTorch device during the scan.

## Gradient Readiness

The core PyTorch scan implementation should avoid unnecessary `.detach()`, `.cpu()`, or NumPy conversion.

This would allow future Machine Learning workflows to keep results on GPU and preserve autograd support.

## Expected Behavior

For `states` scans:

- Select states using the current logic from `STMutils.run_stm_scan(...)`.
- Evaluate all selected states in one PyTorch scan call.
- Return `didv` with shape:
```python
(n_states, nx, ny, nz)
```
For `STM(...)`:

- Build voltage values with the same semantics as the current implementation.
- Compute all `dI/dV` values in PyTorch.
- Integrate current in PyTorch.
- Return the same result shape as the current `STM(...)`.

For `MSTM(...)` / voltage scans:

- Build voltage values with the same semantics as the current implementation.
- Compute the full `dIdVs` stack in PyTorch.
- Integrate `STMs` from `dIdVs` using PyTorch operations.
- Return:
```python
STMs, dIdVs
```
with shapes compatible with the current API.

## Acceptance Criteria

- `scan_type in ["states", "STATES"]` uses the PyTorch scan-level implementation when `didv_backend == DidvBackend.PYTORCH`.
- `STM(..., backend=DidvBackend.PYTORCH)` uses the PyTorch scan-level implementation.
- `MSTM(..., backend=DidvBackend.PYTORCH)` uses the PyTorch scan-level implementation.
- State scans avoid per-state CPU/GPU transfers.
- Voltage scans avoid per-voltage CPU/GPU transfers.
- Results match the existing loop-based PyTorch implementation within a documented numerical tolerance.
- Existing APIs remain backward compatible.
- Tests cover:
  - `states` scan vs current loop behavior
  - `STM` scan vs current loop behavior
  - `MSTM` scan vs current loop behavior
  - CPU execution
  - CUDA execution when available
- Add a small benchmark comparing:
  - current Python loop over `dIdV(..., backend=DidvBackend.PYTORCH)`
  - new PyTorch scan-level implementation
